File: TripKitAndroid/src/main/java/com/skedgo/geocoding/agregator/MGAResultInterface.java
package com.skedgo.geocoding.agregator;

import java.io.Serializable;
import java.util.List;

/**
 * this class represents a scored result, it could be single or have duplicates
 */
public interface MGAResultInterface<T extends GCResultInterface> extends Serializable {

    //  returns query result
    T getResult();

    //  total score for the result
    int getScore();

    //  returns all duplicates for the result
    List<MGAResultInterface<T>> getDuplicates();

    //  returns the scored result which is the class representative for the duplicate list
    MGAResultInterface<T> getClassRepresentative();

    //next was added to test
//    name score value
    int getNameScore();

    //    address score value
    int getAddressScore();

    //    distance score value
    int getDistanceScore();

    //    populariry score value
    int getPopularityScore();
}

File: TripKitAndroid/src/main/java/com/skedgo/geocoding/agregator/MGAResultInterface.kt
package com.skedgo.geocoding.agregator

import java.io.Serializable

/**
 * this class represents a scored result, it could be single or have duplicates
 */
interface MGAResultInterface<T : GCResultInterface> : Serializable {

    // Returns query result
    val result: T

    // Total score for the result
    val score: Int

    // Returns all duplicates for the result
    val duplicates: List<MGAResultInterface<T>>?

    // Returns the scored result which is the class representative for the duplicate list
    val classRepresentative: MGAResultInterface<T>?

    // Name score value
    val nameScore: Int

    // Address score value
    val addressScore: Int

    // Distance score value
    val distanceScore: Int

    // Popularity score value
    val popularityScore: Int
}
